setup.py

```diff
@@ -55,10 +55,10 @@
         "comment_parser",
         "beautifulsoup4",
         "mwparserfromhell @ git+https://github.com/earwig/mwparserfromhell.git@0f89f44",
-        "cudf-cu12==23.10.*",
-        "dask-cudf-cu12==23.10.*",
-        "cugraph-cu12==23.10.*",
-        "dask-cuda==23.10.*",
+        "cudf-cu12==23.12.*",
+        "dask-cudf-cu12==23.12.*",
+        "cugraph-cu12==23.12.*",
+        "dask-cuda==23.12.*",
```

Comment:
I think we want to bump this way up to `24.02` which is the latest stable RAPIDS release.  (See release blog [link](https://medium.com/rapids-ai/rapids-24-02-release-43ad88db0627))

```suggestion
        "cudf-cu12==24.02.*",
        "dask-cudf-cu12==24.02.*",
        "cugraph-cu12==24.02.*",
        "dask-cuda==24.02.*",
```

Reply:
Is there a clean way to pull of a lower bound on the RAPIDS libraries so we don't have to manually update the versions? Not sure if that's good practice though. In the "Remove wildcard suffix" commit I tried to do it but got these errors:
```
cugraph 23.12.0 requires dask-cuda==23.12.*, but you have dask-cuda 24.2.0 which is incompatible.
cugraph 23.12.0 requires rapids-dask-dependency==23.12.*, but you have rapids-dask-dependency 24.2.0 which is incompatible.
cugraph-service-server 23.12.0 requires dask-cuda==23.12.*, but you have dask-cuda 24.2.0 which is incompatible.
cugraph-service-server 23.12.0 requires rapids-dask-dependency==23.12.*, but you have rapids-dask-dependency 24.2.0 which is incompatible.
cuml 23.12.0 requires dask-cuda==23.12.*, but you have dask-cuda 24.2.0 which is incompatible.
cuml 23.12.0 requires rapids-dask-dependency==23.12.*, but you have rapids-dask-dependency 24.2.0 which is incompatible.
dask-cudf 23.12.0 requires rapids-dask-dependency==23.12.*, but you have rapids-dask-dependency 24.2.0 which is incompatible.
raft-dask 23.12.0 requires dask-cuda==23.12.*, but you have dask-cuda 24.2.0 which is incompatible.
raft-dask 23.12.0 requires rapids-dask-dependency==23.12.*, but you have rapids-dask-dependency 24.2.0 which is incompatible
```

Reply:
Unsure why your lower bound commit failed, trying again locally to see if i can repro.

Reply:
FWIW, below seems to be working with base environment `python=3.10`

Currently it seems like for RAPIDS pip installation  is supported in  Python 3.9 and 3.10. 
The nightly builds also support 3.11, so in the next release that should work too. 

<img width="988" alt="image" src="https://github.com/NVIDIA/NeMo-Curator/assets/4837571/00e27740-e04e-4cb4-a7ea-cf50a8d38414">


```diff
diff --git a/setup.py b/setup.py
index ae51575..0f4fc95 100644
--- a/setup.py
+++ b/setup.py
@@ -36,7 +36,7 @@ setup(
         "Programming Language :: Python :: 3",
     ],
     packages=find_packages(),
-    python_requires=">=3.7",
+    python_requires=">=3.10",
     install_requires=[
         "dask[complete]>=2021.7.1",
         "distributed>=2021.7.1",
@@ -55,15 +55,16 @@ setup(
         "comment_parser",
         "beautifulsoup4",
         "mwparserfromhell @ git+https://github.com/earwig/mwparserfromhell.git@0f89f44",
-        "cudf-cu12==23.10.*",
-        "dask-cudf-cu12==23.10.*",
-        "cugraph-cu12==23.10.*",
-        "dask-cuda==23.10.*",
+        "cudf-cu12>=24.2",
+        "dask-cudf-cu12>=24.2",
+        "cugraph-cu12>=24.2",
+        "dask-cuda>=24.2",
         "spacy>=3.6.0, <4.0.0",
         "presidio-analyzer==2.2.351",
         "presidio-anonymizer==2.2.351",
         "usaddress==0.5.10",
         "nemo_toolkit[nlp]>=1.23.0",
+       
     ],
     entry_points={
         "console_scripts": [
```

Reply:
I think we should set `python_requires=3.10` and then once rapids releases (which should be on  Wed, Apr 12, 2023)we can bump it to >=.  
	 
https://github.com/rapidsai/build-planning/issues/3